ci: Add workflow to sync finch submodules

vsiravar opened the pull request:
Issue #, if available: N/A

*Description of changes:*
Add workflow to update submodules in `finch`. 

#### Details
1. Workflow runs daily at 9am UTC. 
2. Has a `workflow_dispatch` trigger for manually running the workflow in addition to the regular daily cadence. 
*Testing done:*
Yes. See https://github.com/vsiravar/finch-public/pull/2
Test with updating `FINCH_OS_BASENAME` : https://github.com/vsiravar/finch-public/pull/4/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52


- [X] I've reviewed the guidance in CONTRIBUTING.md


#### License Acceptance

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

davidhsingyuchen reviewed (commented):
From #255, it seems that `FINCH_OS_BASENAME` has to be in sync with what's under `finch-core/hashes`. I'm not very sure about this, but maybe it makes sense to also update `FINCH_OS_BASENAME` when updating submodules? /cc @pendo324

vsiravar commented:
> From #255, it seems that `FINCH_OS_BASENAME` has to be in sync with what's under `finch-core/hashes`. I'm not very sure about this, but maybe it makes sense to also update `FINCH_OS_BASENAME` when updating submodules? /cc @pendo324

Makes sense, `finch-core` uses [sha512 to check os base image file integrity ](https://github.com/runfinch/finch-core/blob/main/Makefile#L51) while in the [finch.yaml file we use sha256](https://github.com/runfinch/finch/blob/main/Makefile#L47). I guess that's the reason why we need  `FINCH_OS_BASENAME` in finch's Makefile. Added script to update `FINCH_OS_BASENAME` in the Makefile along with updating submodules. Added test in the description.  cc @davidhsingyuchen @pendo324

pendo324 reviewed (approved):
Yeah, the comments in the PR are correct; the name has to be in sync. I'm actually wondering if its possible to remove this condition, but for now, this action looks good 👍

davidhsingyuchen commented on .github/bin/update-os-image.sh:
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+set -x
```
More info: https://coderwall.com/p/fkfaqq/safer-bash-scripts-with-set-euxo-pipefail

```suggestion
set -euxo pipefail
```

davidhsingyuchen commented on .github/workflows/sync-submodules.yaml:
```diff
@@ -0,0 +1,29 @@
+        uses: peter-evans/create-pull-request@v4
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          commit-message: bump submodules
```
nit: Since it doesn't add additional info given the title below, maybe we can just drop it

```suggestion
```
